onboarding/source/amb_light_service.c

```diff
@@ -26,14 +59,58 @@ uint8_t initLightService(void) {
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+
+    light_event_t event;
+    while(1) {
+        if (xQueueReceive(xQueue1, &event, (TickType_t) 0) == pdPASS) {
+            if(event == MEASURE_LIGHT) {
+                // if event recieved is MEASURE_LIGHT, get value from adc and send over serial
+                uint8_t data = getAmbientLightData();
+                sciPrintText(scilinREG, &data, sizeof(uint8_t));
+            }
+        }
+    }
 
     /* USER CODE END */
 }
 
 uint8_t sendToLightServiceQueue(light_event_t *event) {
     /* USER CODE BEGIN */
     // Send the event to the queue.
-    
+    if (xQueueSend(xQueue1, (void *) &event, (TickType_t) 0) == pdPASS) {
+        return 1;
+    }
     /* USER CODE END */
     return 0;
 }
+
+// Function that does ADC conversion and returns ambient light value
+uint8_t getAmbientLightData(void) {
+    adcData_t adc_data;
+    adcData_t *adc_data_ptr = &adc_data;
+
+    adcStartConversion_selChn(adcREG1, Light_Sensor, 6, adcGROUP1);
+
+    while(!adcIsConversionComplete(adcREG1, adcGROUP1));
+
+    adcGetSingleData(adcREG1, adcGROUP1, adc_data_ptr);
+
+    return(adc_data_ptr->value);
+}
+
+void adcGetSingleData(adcBASE_t *adc, unsigned group, adcData_t *data) {
```

Comment:
If you're interested, an alternate way to do this ADC conversion is using the included ADC HAL functions which can be found in `hal/source/adc.c`. Some functions that would have been useful are:
- `adcStartConversion`
- `adcGetData`

I'm mentioning this for future reference since it's easier to use the HAL functions than to basically re-implement them. Also, the adc.c file is pretty well-commented, so if you'd like to understand some of this ADC code a bit more, you can check it out. For example, the reason we & the buf with 0xFFFU is because the RM46 ADC modules are 12-bit.